testing/testcafe/tests/scheduler/agenda/keyField.ts

```diff
@@ -0,0 +1,145 @@
+
+['week', 'agenda'].forEach((currentView) => {
+  test(`Waring should be throw in console in case currentView='${currentView}'(T1100758)`, async (t) => {
+    const messages = await t.getBrowserConsoleMessages();
+
+    const isWarningExist = !!messages.warn.find(hasWarningCode);
+    await t.expect(isWarningExist).ok();
+  }).before(async () => {
+    await createWidget('dxScheduler', {
+      dataSource: [],
+      views: ['week', 'agenda'],
+      currentView,
+      currentDate: new Date(2021, 2, 28),
+      height: 600,
+    });
+  });
+});
+
+test('Waring should be throw in console after set new views(T1100758)', async (t) => {
+  const messages = await t.getBrowserConsoleMessages();
+  const isWarningExist = !!messages.warn.find(hasWarningCode);
+  await t.expect(isWarningExist).notOk();
+
+  const scheduler = new Scheduler('#container');
+  await scheduler.option('views', ['week', 'agenda']);
+
+  const messagesAfterChangeViews = await t.getBrowserConsoleMessages();
+  const isWarningExistAfterChangeViews = !!messagesAfterChangeViews.warn.find(hasWarningCode);
+  await t.expect(isWarningExistAfterChangeViews).ok();
+}).before(async () => {
+  await createWidget('dxScheduler', {
+    dataSource: [],
+    views: ['week'],
+    currentView: 'week',
+    currentDate: new Date(2021, 2, 28),
+    height: 600,
+  });
+});
+
+// TODO
+// The matrix test to split, so it was not possible to achieve a working state matrix test
+
+test('Waring shouldn\'t be throw in console in case currentView=\'week\' if keyField exists(T1100758)', async (t) => {
+  const messages = await t.getBrowserConsoleMessages();
+
+  const isWarningExist = !!messages.warn.find(hasWarningCode);
+  await t.expect(isWarningExist).notOk();
+}).before(async () => {
+  await createWidget('dxScheduler', () => {
+    const store = new (window as any).DevExpress.data.CustomStore({
+      key: 'id',
+      load: () => [],
+    });
+
+    return {
+      dataSource: store,
+      views: ['week', 'agenda'],
+      currentView: 'week',
+      currentDate: new Date(2021, 2, 28),
+      height: 600,
+    };
+  });
+});
+
+test('Waring shouldn\'t be throw in console in case currentView=\'agenda\' if keyField exists(T1100758)', async (t) => {
+  const messages = await t.getBrowserConsoleMessages();
+
+  const isWarningExist = !!messages.warn.find(hasWarningCode);
+  await t.expect(isWarningExist).notOk();
+}).before(async () => {
+  await createWidget('dxScheduler', () => {
+    const store = new (window as any).DevExpress.data.CustomStore({
+      key: 'id',
+      load: () => [],
+    });
+
+    return {
+      dataSource: store,
+      views: ['agenda'],
+      currentView: 'agenda',
+      currentDate: new Date(2021, 2, 28),
+      height: 600,
+    };
+  });
+});
+
+//
+
+['week', 'agenda'].forEach((currentView) => {
+  test(`Waring should be throw in console in case currentView='${currentView}' if keyField not set in Store(T1100758)`, async (t) => {
```

Comment:
```suggestion
  test(`Warning should be thrown in console in case currentView='${currentView}' if keyField not set in Store(T1100758)`, async (t) => {
```